### PR TITLE
Show real constraint summary in obs badge

### PR DESCRIPTION
**Pull request by toddburnside**

### CODE REVIEW

**Comment by cquiroz** [comment]
Do you have a screenshot?

File: model/shared/src/main/scala/explore/model/ConstraintsSummary.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  def summaryString: String =
+    s"${imageQuality.label} ${cloudExtinction.label} ${skyBackground.label} ${waterVapor.label}"
+}
+
+object ConstraintsSummary {
+
+  implicit val constraintSummaryDecoder: Decoder[ConstraintsSummary] = deriveDecoder
+  implicit val constraintSummaryEncoder: Encoder[ConstraintsSummary] = deriveEncoder
+
+  implicit val eqConstraintSummary: Eq[ConstraintsSummary] = Eq.fromUniversalEquals
```
Comment (cquiroz) [inline model/shared/src/main/scala/explore/model/ConstraintsSummary.scala:37]: We may want to make these from other `Eq` instances doing e.g. `Eq.by` unless all members have trivial `Eq` instances

File: model/shared/src/main/scala/explore/model/ObsSummary.scala [hunk 2]
```diff
@@ -26,27 +25,35 @@ import java.time.temporal.ChronoUnit
...
-  conf:            String = "GMOS-N R831 1x300",
-  constraints:     String = "<0.7\" <0.3 mag Bright",
-  duration:        Duration = Duration.of(93, ChronoUnit.MINUTES),
-  pointingId:      Option[PointingId],
-  constraintSetId: Option[ConstraintSet.Id]
-)
+  id:          Observation.Id,
+  name:        Option[NonEmptyString],
+  status:      ObsStatus = ObsStatus.New,
+  constraints: Option[ConstraintsSummary],
```
Comment (cquiroz) [inline model/shared/src/main/scala/explore/model/ObsSummary.scala:31]: Nice

**Review by cquiroz — APPROVED** [review]
Great progress

**Review by rpiaggio — APPROVED** [review]
Nice. LGTM.

**Comment by toddburnside** [comment]
> Do you have a screenshot?

With contraints:
image
No constraints:
image

File: model/shared/src/main/scala/explore/model/ConstraintsSummary.scala (hunk 1 above)
Comment (toddburnside) [reply model/shared/src/main/scala/explore/model/ConstraintsSummary.scala]: Switched to `Eq.by`. I hope Eq and Reusability support the new deriving in Scala 3. I don't like how they can get out of date without warning. I found and instance of that for `Reusabilty.by` recently.

**Pull request merged**